<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DesUi</class>
 <widget class="QTabWidget" name="DesUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TabWidget</string>
  </property>
  <property name="currentIndex">
   <number>1</number>
  </property>
  <widget class="QWidget" name="tab">
   <attribute name="title">
    <string>DES加密演示</string>
   </attribute>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>30</y>
      <width>58</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>64位明文</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>30</y>
      <width>58</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>64位密钥</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>280</y>
      <width>58</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>64位密文</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_mText">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>60</y>
      <width>113</width>
      <height>26</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_Key">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>60</y>
      <width>113</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_cText">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>280</y>
      <width>201</width>
      <height>26</height>
     </rect>
    </property>
    <property name="echoMode">
     <enum>QLineEdit::Normal</enum>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_Encrypt">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>60</y>
      <width>60</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>加密</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_IpSub">
    <property name="geometry">
     <rect>
      <x>169</x>
      <y>125</y>
      <width>91</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>初始置换</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_Iter16">
    <property name="geometry">
     <rect>
      <x>169</x>
      <y>165</y>
      <width>91</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>十六轮加密变换</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_InvIpSub">
    <property name="geometry">
     <rect>
      <x>169</x>
      <y>205</y>
      <width>91</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>终结置换</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_KeyGen">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>130</y>
      <width>80</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>密钥生成演示</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="tab_2">
   <attribute name="title">
    <string>加解密</string>
   </attribute>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>120</y>
      <width>58</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>明文</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>260</y>
      <width>58</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>密文</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_mText2">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>120</y>
      <width>301</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_cText2">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>260</y>
      <width>301</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>50</y>
      <width>411</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>64位密钥</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_Key2"/>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_Encrypt_2">
       <property name="text">
        <string>加密</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_Decrypt_2">
       <property name="text">
        <string>解密</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
